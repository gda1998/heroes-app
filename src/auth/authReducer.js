import { types } from '../types/types';

// const state = {
//     name: 'Gabriel',
//     logged: true
// };

/**
 * Metodo reducer para login y logout de la app
 * 
 * @param  {JSON} state={}
 * @param  {JSON} action
 * @returns  {JSON}
 */
export const authReducer = ( state = {}, action ) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            };
        case types.logout:
            return {
                logged: false
            };
        default:
            return state;
    }
}
