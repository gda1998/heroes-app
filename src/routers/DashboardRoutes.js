import { Routes, Route, Navigate } from 'react-router';

import { Navbar } from '../components/ui/Navbar';
import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mt-2">
                <Routes>
                    <Route path="/hero/:heroId" element={ <HeroScreen /> } />
                    <Route path="/marvel" element={ <MarvelScreen /> } />
                    <Route path="/dc" element={ <DcScreen /> } />
                    <Route path="/search" element={ <SearchScreen /> } />
                    <Route path="/" element={ <MarvelScreen /> } />
                    <Route path="/*" element={ <Navigate to="/" replace /> } />
                </Routes>
            </div>   
        </>
    )
}
