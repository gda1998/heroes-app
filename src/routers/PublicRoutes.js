import { useContext } from 'react';
import { Navigate } from 'react-router';
import { AuthContext } from '../auth/AuthContext';

export const PublicRoutes = ({ children }) => {
    const { user } = useContext(AuthContext);
    return user.logged ? <Navigate to="/" /> : children;
}