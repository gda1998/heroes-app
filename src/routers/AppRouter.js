import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>

                <Route path="/login" element={ 
                    <PublicRoutes>
                        <LoginScreen />        
                    </PublicRoutes>
                } />
                <Route path="/*" element={
                    <PrivateRoutes>
                        <DashboardRoutes />
                    </PrivateRoutes>
                } />

            </Routes>
        </BrowserRouter>
    );
}
