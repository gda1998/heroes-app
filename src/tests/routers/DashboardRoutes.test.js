import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('Pruebas en <DashboardRoutes />', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Gabriel',
            logged: true
        }
    };

    // beforeEach( () => {
    //     jest.clearAllMocks();
    // });

    test('Debe de mostrarse correctamente', () => {
        const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter>
                <DashboardRoutes />
            </MemoryRouter>
        </AuthContext.Provider>);
        const userName = wrapper.find('.text-name').text().trim();
        expect(wrapper).toMatchSnapshot();
        expect(userName).toBe('Gabriel');
    });
});