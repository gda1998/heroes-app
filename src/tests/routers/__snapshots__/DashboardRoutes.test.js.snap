// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <button
        aria-controls="navbarTogglerDemo01"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarTogglerDemo01"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarTogglerDemo01"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Asociaciones
        </a>
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link "
              href="/marvel"
              onClick={[Function]}
            >
              Marvel
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link "
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link "
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
        </ul>
        <div
          className="d-flex"
        >
          <span
            className="text-name mx-3 my-2 nav-item"
          >
            Gabriel
          </span>
          <button
            className="btn link-logout"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    Array [
      <h1>
        Marvel Heroes
      </h1>,
      <hr />,
      <div
        className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
      >
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  loading="lazy"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
           
        </div>
      </div>,
    ]
  </div>,
]
`;
